CanvasColorSpaceProposal.md

```diff
@@ -63,56 +104,72 @@ partial interface CanvasRenderingContext2D {
-* Note that pixel values written to WebGL draw buffers are now in a space with a linear transfer function, and will not necessarily match those of an "srgb" canvas.
-
-#### The pixelFormat context creation attribute
-The pixelFormat attributes specifies the numeric types to be used for storing pixel values.
-* Support for "8-8-8-8" is mandatory. All other formats ar optional.
-* When an unsupported format is requested, the format shall fall back to "8-8-8-8".
-* With formats that use integer numeric types for the color channels, the canvas pixel buffer shall store non-linear color values, encoded using the transfer curves prescribed by the specification of the current color space.
-* With formats that use floating-point numeric types for the color channels, the canvas pixel buffer shall store linear (i.e unencoded) color values.
-* Float values outside of [0,1] range can be used to represent colors outside of the chosen color gamut. This allows float pixel formats to represent all possible colors and brightness levels. How values outside of [0,1] are displayed depends on the capabilites of the device and output display. Some implementations may simply clamp these values to [0,1]. If the device and display are capable, a pixel value of (2,2,2) should be twice as bright as (1,1,1).
+* toDataURL/toBlob produce resources tagged as being in the sRGB color space, if the encoding format supports colorspace tagging or embedded color profiles.
+    * Exception: User agents that implement the legacy behavior must not encode any color space metadata.
+
+##### The "rec-2020" color space
+* As per the [CSS rec-2020 color space](https://www.w3.org/TR/css-color-4/#valdef-color-rec-2020).
+* This color space has different primaries and a different transfer function than "srgb".
+* Support is optional, and should not be present on srgb-only UAs.
+
+
+#### The colorEncoding context creation attribute
+The colorEncoding attributes specifies the encoding to be used for storing pixel channel color values.
+* Support for "unorm8" is mandatory. All other encodings are optional.
+* When an unsupported encoding is requested, the encoding shall fall back to "unorm8".
+* The alpha channel is always interpreted as if clamped to [0,1].
+* Float RGB channel values outside of [0,1] range can be used to represent colors outside of the chosen color gamut. This allows float pixel formats to represent all possible colors and brightness levels. How values outside of [0,1] are displayed depends on the capabilites of the device and output display. Some implementations may simply clamp these values to [0,1]. If the device and display are capable, a luminance-linear pixel value of (2,2,2) should be twice as bright as (1,1,1).
+* Operations on encoded values always operate on decoded values, not the encoded bits.
+    * I.e. in "unorm8-srgb" encoding, 0xff (1.0) minus 0xbc (0.5) equals 0xbc (0.5).
+
+
+#### The colorLinearity context creation attribute
```

Comment:
The spec should probably give more information on how this field is expected to be utilized/implemented.

Reply:
I agree. It's not clear to me how it would interact with a choice of unorm8 vs. unorm8-srgb encoding, for example. How about making the other changes first and deferring addition of the linearity enum?

Reply:
Happy to do that, though the linearity enum was the replacement for the behavior of the previous version's rec2020 enum, at least to my understanding of it.
I do think that it's non-essential, but I included it for completeness.

Reply:
I too do not understand the interaction.

As mentioned earlier in my review (and Alexey mentioned too) I think separating primaries and transfer function will be helpful. For example some proposals for handling wider encodings use sRGB/BT.709 primaries with a linear transfer function. We separated these in the Khronos Data Format Specification> I think it has help simplify things because of the extra flexibility it offers.